### PR TITLE
[Needs-Carry] Update opencontainers/runtime-spec to v1.0.0

### PR DESCRIPTION
**Pull request by darstahl**
This needs to coordinate with https://github.com/moby/moby/pull/34356

**- What I did**

In order to prepare for the update to containerd, Docker needs to update to the v1.0.0 opencontainers/runtime-spec. This removes all out of band fields on Windows, and getting this done now reduces my workload on LCOW bringup in containerd by a substantial amount.

I'm not familiar with the Linux and/or Solaris impact here. I did enough to get LCOW passing basic validation tests, but did not check if Linux compiles or needs changes (It almost certainly will with at least the Process/Box pointer changes)

**- How I did it**

Update runtime-spec and change Windows to use the new fields.

**- How to verify it**

Manually verified as much as possible. CI to confirm.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

My dog

/cc @dmcgowan @mlaventure @crosbymichael Can any of you carry this with the Linux changes?

/cc @jhowardmsft This will likely cause some merge conflicts with ongoing LCOW work FYI

/cc @jstarks FYI

### CODE REVIEW

**Comment by mlaventure** [comment]
@darrenstahlmsft I just fixed a panic in my branch, you may want to rebase so CI can run with your changes.

**Comment by mlaventure** [comment]
@darrenstahlmsft Just realized you've updated the runtime-spec to `v1.0.0` in this branch.

I guess we would have to pass by it sooner than later. Going to have to do the same for `containerd`

**Comment by darrenstahlmsft** [comment]
@mlaventure both versions are fine for me. The only difference is that specs.Root needs to be populated on v1.0.0. I have a version based off your PR that also works.

**Comment by darrenstahlmsft** [comment]
Though updating needs to happen eventually anyway.

**Comment by darrenstahlmsft** [comment]
I've validated this change against LCOW with some private updates. Current status is that containerd v0.2.x needs an update to v1.0.0 of runtime-spec, then this needs to be updated with the Linux changes, or carried in another PR.

File: libcontainerd/client_windows.go [hunk 1]
```diff
@@ -305,21 +320,21 @@ func (clnt *client) createLinux(containerID string, checkpoint string, checkpoin
...
-		}
+	if spec.Windows == nil {
+		return fmt.Errorf("spec.Windows must not be nil for LCOW containers")
 	}
 
-	// We must have a layer option with at least one path
-	if layerOpt == nil || layerOpt.LayerPaths == nil {
-		return fmt.Errorf("no layer option or paths were supplied to the runtime")
+	// We must have least two layers in the spec, the bottom one being a base image,
+	// the top one being the RW layer.
```
Comment (jhowardmsft) [inline libcontainerd/client_windows.go]: I don't believe this is true for LCOW

**Comment by mlaventure** [comment]
I've made a PR to updated containerd to runtime-spec v1.0.0 [here](https://github.com/containerd/containerd/pull/1281)

File: libcontainerd/client_windows.go (hunk 1 above)
Comment (mlaventure) [inline libcontainerd/client_windows.go]: On the old PR, the minimum seems to be 1

Comment (jhowardmsft) [inline libcontainerd/client_windows.go]: That is correct, it was. But two is correct (for Windows containers, not lcow) in the option-free mode. We combined a field into the layer folders

**Comment by mlaventure** [comment]
It conflicts again :sweat_smile:

**Comment by darrenstahlmsft** [comment]
Change is big enough that everything conflicts 😅 

Updated (and fixed an auto merge issue causing build failure)

**Pull request closed without merging**